scico/test/test_data.py

```diff
@@ -1,13 +1,23 @@
+import os
+
 from scico import data
 
 
 class TestSet:
     def test_kodim23_uint(self):
+        if len(os.listdir(os.path.abspath("./data"))) == 0:
+            raise IOError(
+                "\nThe data submodule must be cloned and initialized. If already cloned use the following in the root directory:\n\tgit submodule update --init --recursive\nOtherwise, make sure to clone using:\n\tgit clone --recurse-submodules git@github.com:lanl/scico.git\nAnd after cloning run:\n\tgit submodule init && git submodule update."
+            )
         x = data.kodim23()
```

Comment:
It is, but this way, the other two tests don't run and give the other error which has no information as to what's happening. Maybe I can expand a bit on the error message if I use your approach and mention the rest of the cases might not run? I'm currently looking at a way to stop this particular test class completely if the data module isnt available, and throw the appropiate error to pytest. We can leave it on hold for now, but this can be a good starting point.